src/main/java/com/uber/cadence/internal/replay/ReplayDecider.java

```diff
@@ -435,15 +437,84 @@ private void decideImpl(PollForDecisionTaskResponse decisionTask, Functions.Proc
     }
   }
 
+  private boolean processEventLoop(long startTime, int decisionTimeoutSecs, DecisionEvents decision)
+      throws Throwable {
+    eventLoop();
+
+    if (decision.isReplay()) {
+      return replayLocalActivities(decision);
+    } else {
+      return executeLocalActivities(startTime, decisionTimeoutSecs, decision);
+    }
+  }
+
+  private boolean replayLocalActivities(DecisionEvents decision) throws Throwable {
+    List<HistoryEvent> localActivityMarkers = new ArrayList<>();
+    for (HistoryEvent event : decision.getMarkers()) {
+      if (event
+          .getMarkerRecordedEventAttributes()
+          .getMarkerName()
+          .equals(ClockDecisionContext.LOCAL_ACTIVITY_MARKER_NAME)) {
+        localActivityMarkers.add(event);
+      }
+    }
+
+    if (localActivityMarkers.isEmpty()) {
+      return false;
+    }
+
+    int processed = 0;
+    while (context.numPendingLaTasks() > 0) {
+      int numTasks = context.numPendingLaTasks();
+      for (HistoryEvent event : localActivityMarkers) {
+        processEvent(event);
+      }
+
+      eventLoop();
+
+      processed += numTasks;
+      if (processed == localActivityMarkers.size()) {
+        return false;
+      }
+    }
+    return false;
+  }
+
+  private boolean executeLocalActivities(
+      long startTime, int decisionTimeoutSecs, DecisionEvents decision)
+      throws InterruptedException {
+    while (context.numPendingLaTasks() > 0) {
+      context.startUnstartedLaTasks();
+
+      while (context.numPendingLaTasks() > 0) {
+        Thread.sleep(100);
+      }
+
+      eventLoop();
+
+      if (context.numPendingLaTasks() == 0) {
+        return false;
+      }
+
+      // Break local activity processing loop if we almost reach decision task timeout.
```

Comment:
Add comment explaining that breaking out of the loop doesn't mean that the decision or local activities are aborted due to immediately created decision task.